pandera/engines/pandas_engine.py

```diff
@@ -1330,3 +1330,39 @@ def __init__(  # pylint:disable=super-init-not-called
 
     def __str__(self) -> str:
         return str(NamedTuple.__name__)
+
+
+@Engine.register_dtype(equivalents=[UnionType, Union, "UnionType", "union"])
+@dtypes.immutable(init=True)
+class PythonUnion(PythonGenericType):
+    """A datatype to support python generics."""
+
+    type = UnionType
+
+    def __init__(  # pylint:disable=super-init-not-called
+        self, generic_type: Optional[Type] = None
+    ) -> None:
+        if generic_type is not None:
+            object.__setattr__(self, "generic_type", generic_type)
+
+    def check(
+        self,
+        pandera_dtype: dtypes.DataType,
+        data_container: Optional[PandasObject] = None,
+    ) -> Union[bool, Iterable[bool]]:
+        """Check that data container has the expected type."""
+        pandera_dtype = Engine.dtype(pandera_dtype)
+
+        pandas_types = [object]
+        pandas_types.extend(
+            self.generic_type.__args__  # pylint: disable=no-member
+        )
+
+        # the underlying pandas dtype must be an object
+        if pandera_dtype not in map(Engine.dtype, pandas_types):
+            return False
+
+        if data_container is None:
```

Comment:
@karajan1001 can we add a test case for when `data_container is None`?

Reply:
Hello, @cosmicBboy .I'm sorry I do not know how to create a test case in which data_container is None.
I tried 
"""
pd.DataDrame{[]}
"""
But it just put an empty `pd.Series` in.